### DISCUSSION TITLE
Adding the ability to seed or parse ICS files

### DISCUSSION THREAD

**Discussion by 99linesofcode** [spatie/icalendar-generator#70] [Ideas]
Hi,

While I recognize that this package is probably intentionally named a generator, I was wondering whether you have considered adding support for parsing existing `.ics` files in the past.

Sure, there are other packages out there that provide similar functionality but non that have such an elegant and Fluent syntax which so closely resembles Laravel's.

What are your thoughts on this?

**Comment by rubenvanassche** [comment]
Hi @99linesofcode,

At the moment we do not have such plans to support parsing ics files. If it was something simple we could do with a few lines of code then I would add it immediately but this would require a complete rewrite and rethinking of the package API which is a bit too much for such a feature.

I haven't used `.ics` parsers, so I don't know what kind of packages exist. But you could maybe build repositories around the output from another package that enable a fluent syntax?

Good luck!
